packages/pluggableWidgets/pie-doughnut-chart-native/src/utils/SeriesLoader.ts

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from "react";
+import { ensure } from "@mendix/pluggable-widgets-tools";
+
+import { SeriesType } from "../../typings/PieDoughnutChartProps";
+import { Slice, DataPoints } from "../components/PieDoughnutChart";
+
+interface ChartSeries {
+    slices: DataPoints;
+}
+
+export function useSeries(series: SeriesType[]): ChartSeries[] | null {
+    const [chartSeries, setChartSeries] = useState<ChartSeries[] | null>(null);
+
+    useEffect(() => {
+        const loadedSeries: ChartSeries[] = [];
+        for (const element of series) {
+            const result = loadStaticSeries(element);
+            if (!result) {
+                setChartSeries(null);
+                return;
+            }
+            loadedSeries.push(result);
+        }
+        setChartSeries(loadedSeries);
```

Comment:
I think you could make this shorter;
```js
setChartSeries(series.map(element => loadStaticSeries(element) ?? null).filter(element => !!element))
```

Reply:
Hmm, I tried to implement this but it changes the logic;

The hook `useSeries`'s return signature is `ChartSeries[] | null`, representative of a proper, complete dataset _or_ `null` when not.

The expression you suggest has a return signature of `ChartSeries[]`.

I could do the following: `setChartSeries(series.map(_series => loadStaticSeries(_series) ?? null)` then in the calling component, check if _any_ of the items is null with `some`. What do you think?

Reply:
It does mean shorter code, although computationally, more expensive, since the current implementation would exit early if improper data is found, whereas this suggested approach would continue till all elements processed.

Reply:
Hmm yeah performance is more important. Lets keep the for loop here